datalad/support/network.py

```diff
@@ -1103,4 +1152,60 @@ def download_url(url, dest=None, overwrite=False):
         return providers.fetch(url)
 
 
+def local_path2url_path(local_path: str,
+                        auto_resolve: bool = False
+                        ) -> str:
+    """Convert a local path into a URL path component"""
```

Comment:
Describe  auto_resolve. Also code uses only with False, so why bother having this option at all? Just document that path must be empty and done?!

Reply:
I changed `auto_resolve` to `allow_relative_path`. The path can be empty if `allow_relative_path` is `True`. That actually leads to some more simplifications.

Reply:
why bother having it if we don't use it in the code?

Reply:
We use it.